From d307a38515c5d050bbf8555d79aff3e4ce78af78 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Mon, 25 May 2015 12:08:47 -0400
Subject: [PATCH] Gentoo-specific packaging fixes

Fix cross-compiles
EPREFIX support
Revert custom build and install classes
---
 setup.py | 20 ++++----------------
 1 file changed, 4 insertions(+), 16 deletions(-)

diff --git a/setup.py b/setup.py
index fc89513..806464f 100644
--- a/setup.py
+++ b/setup.py
@@ -81,29 +81,19 @@ class _M2CryptoBuildExt(build_ext.build_ext):
 
         build_ext.build_ext.finalize_options(self)
 
-        self.add_multiarch_paths()
-
         includeDir = os.path.join(self.openssl, 'include')
         opensslIncludeDir = os.path.join(self.openssl, 'include', 'openssl')
         opensslLibraryDir = os.path.join(self.openssl, 'lib')
 
-        self.swig_opts = ['-I%s' % i for i in self.include_dirs + \
-                          [opensslIncludeDir, includeDir]]
+        eprefix = os.getenv('EPREFIX', '')
+        self.swig_opts = ['-I' + eprefix + '/usr/include']
         self.swig_opts.append('-includeall')
         self.swig_opts.append('-modern')
 
-        # Fedora does hat tricks.
-        if platform.linux_distribution()[0] in ['Fedora', 'CentOS']:
-            if platform.architecture()[0] == '64bit':
-                self.swig_opts.append('-D__x86_64__')
-            elif platform.architecture()[0] == '32bit':
-                self.swig_opts.append('-D__i386__')
-
         self.swig_opts.append('-outdir')
         self.swig_opts.append(os.path.join(os.getcwd(),'M2Crypto'))
 
-        self.include_dirs += [os.path.join(self.openssl, opensslIncludeDir),
-                              os.path.join(os.getcwd(), 'SWIG')]
+        self.include_dirs += [os.path.join(os.getcwd(), 'SWIG')]
 
         if sys.platform == 'cygwin':
             # Cygwin SHOULD work (there's code in distutils), but
@@ -113,8 +103,6 @@ class _M2CryptoBuildExt(build_ext.build_ext):
             # Someday distutils will be fixed and this won't be needed.
             self.library_dirs += [os.path.join(self.openssl, 'bin')]
 
-        self.library_dirs += [os.path.join(self.openssl, opensslLibraryDir)]
-
 if sys.platform == 'darwin':
    my_extra_compile_args = ["-Wno-deprecated-declarations"]
 else:
@@ -162,5 +150,5 @@ interface.''',
 
       ext_modules = [m2crypto],
       test_suite='tests.alltests.suite',
-      cmdclass = {'build': CustomBuild, 'install': CustomInstall, 'build_ext': _M2CryptoBuildExt}
+      cmdclass = {'build_ext': _M2CryptoBuildExt}
       )
-- 
2.4.1

